ppb_vector/vector2.py

```diff
@@ -183,14 +177,12 @@ def __repr__(self: VectorOrSub) -> str:
         return f"{type(self).__name__}({self.x}, {self.y})"
 
     def __eq__(self: VectorOrSub, other: typing.Any) -> bool:
-        if is_vector_like(other):
+        try:
             other = Vector2.convert(other)
             return self.x == other.x and self.y == other.y
```

Comment:
I would normally do this in the form:

```
try:
    other = Vector2.convert(other)
except ...:
    return NotImplemented
else:
    return self.x == other.x and self.y == other.y
```

But on reflection, in situations where the comparison raises an error, I think that returning `NotImplemented` would be the right thing. (Although also indicative of a bug in `.convert()`.)

Reply:
Mmmmh, yes, this captures my intent more accurately *and* I think what is going on is clearer.
Thanks <3